### ISSUE TITLE
🐛 `biome search` fails, tries to treat CSS files as JS?

### ISSUE THREAD

**Issue by Sjlver** [biomejs/biome#5979]
### Environment information

```block
CLI:
  Version:                      2.0.0-beta.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.14.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         yarn/3.6.3

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         /home/[redacted]/biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### What happened?

```
$ yarn biome search '`console.log`'
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: /home/runner/work/biome/biome/crates/biome_parser/src/lib.rs:687:13
Thread Name: biome::workspace_worker_2
Message: could not downcast root node to language biome_js_syntax::syntax_node::JsLanguage
```

This seems to happen for CSS files:

```
/path/to/some/file.ts:319:3 search 
  319 │   console.log("some message!");

/path/to/app/globals.css internalError/panic  INTERNAL
```

### Expected result

`biome search` should gracefully handle files of the wrong type.

It may well throw an error message, although that wouldn't be very user-friendly. It should definitely not panic.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
It seems that our message is completely ignored 😅 

> When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

**Comment by Sjlver** [comment]
I had hoped that this triggers some "ah, of course, we're missing a filetype check here and thus cast CSS to JS" reaction...

But OK, OK... let me overcome my laziness and make a repro for you

**Comment by Sjlver** [comment]
https://github.com/Sjlver/biome-repro-1747046861198

Run ``npx biome search '`console.log`'`` to trigger the problem.
